store/providers/xorm/machine_types.go

```diff
@@ -3,14 +3,16 @@ package xorm
 import (
 	"encoding/json"
 
+	"gomodules.xyz/secrets/types"
 	clusterapi "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
 )
 
 type Machine struct {
 	ID        int64  `xorm:"pk autoincr"`
 	Name      string `xorm:"INDEX NOT NULL"`
-	Data      string `xorm:"text NOT NULL"`
+	Data      []byte `xorm:"blob NOT NULL"`
```

Comment:
Why not store this as `text` ?

Reply:
string of encrypted data can't be stored to database as it generates some unsupported characters. that's why I stored as `[]byte`.